src/index.ts

```diff
@@ -1278,7 +1294,24 @@ async function claimLink({
 	const addressHashBinary = ethers.utils.arrayify(addressHash) // v5
 	config.verbose && console.log('addressHash: ', addressHash, ' addressHashBinary: ', addressHashBinary)
 	const addressHashEIP191 = solidityHashBytesEIP191(addressHashBinary)
-	const signature = signAddress(recipient, keys.privateKey) // sign with link keys
+
+	// sign with link keys
+	let signature: string
+	let claimParams: any[]
+	if (contractVersion == 'v4.2') {
+		signature = await signWithdrawalMessage(
+			chainId,
+			contract.address,
+			depositIdx,
+			recipient,
+			keys.privateKey,
+			false
+		)
+		claimParams = [depositIdx, recipient, signature]
+	} else {
+		signature = await signAddress(recipient, keys.privateKey)
+		claimParams = [depositIdx, recipient, addressHashEIP191, signature]
+	}
```

Comment:
Totally makes sense! I did it slightly different from what you proposed though, so lmk if that's ok with you.

I made `signWithdrawalMessage` to return a fully ready list of claim params. Since otherwise we would have to make the same `if version == 'v4.2'` again to create the claim params from the computed signature, which doesn't sound that nice.

Reply:
![image](https://github.com/peanutprotocol/peanut-sdk/assets/12943235/6fe0f0f1-ed05-45ca-898f-d2eeb94a6b02)

Perfect, very clean

Reply:
❤️ ❤️ ❤️